### DISCUSSION TITLE
Help for --Help

### DISCUSSION THREAD

**Discussion by islogged** [openai/whisper#1609] [General]
Hi everyvody,

I read this: https://github.com/openai/whisper/blob/main/README.md

But where i can find more documented and detailed help for all of these: 

> whisper --Help
> usage: whisper [-h] [--model {tiny.en,tiny,base.en,base,small.en,small,medium.en,medium,large-v1,large-v2,large}] [--model_dir MODEL_DIR] [--device DEVICE]
>                [--output_dir OUTPUT_DIR] [--verbose VERBOSE] [--task {transcribe,translate}]
>                [--language ...]
>                [--temperature TEMPERATURE] [--best_of BEST_OF] [--beam_size BEAM_SIZE] [--patience PATIENCE] [--length_penalty LENGTH_PENALTY]
>                [--suppress_tokens SUPPRESS_TOKENS] [--initial_prompt INITIAL_PROMPT] [--condition_on_previous_text CONDITION_ON_PREVIOUS_TEXT] [--fp16 FP16]
>                [--temperature_increment_on_fallback TEMPERATURE_INCREMENT_ON_FALLBACK] [--compression_ratio_threshold COMPRESSION_RATIO_THRESHOLD]
>                [--logprob_threshold LOGPROB_THRESHOLD] [--no_speech_threshold NO_SPEECH_THRESHOLD] [--threads THREADS]
>                audio [audio ...]

A last question, i saw this answer: https://github.com/openai/whisper/discussions/1589#discussioncomment-6679888

> whisper can only translate non-English to English. Try Google Translate or DeepL.

Is it planned to translate English to non-English soon? and is it requested by people as a new functionality ?

Thanks to All.

**Comment by phineas-pta** [comment]
look carefully README again
![](https://github.com/openai/whisper/assets/37548991/550a1e5c-0848-4c01-ae80-b1ce33c954f9)

**Reply by islogged** [reply]
Hi, thx for ur answser, 
I think u just read the Title and not the Post...

**Reply by phineas-pta** [reply]
what do i miss ?

**Reply by glangford** [reply]
@islogged 
To get help on all of the command line options, type `whisper --help` with a lowercase 'h' as @phineas-pta emphasized in his post. There is no more detailed documentation, other than the README, and the associated research paper. If you have a specific question about how to use whisper, I recommend also searching through the past discussions, there is a lot of useful information on different topics.

**Reply by islogged** [reply]
@phineas-pta
> what do i miss ?

Ok i'm sorry, i understand now:

With --help there is 2 sections "positional arguments:" and "options", which not appear with --Help.
So the Help is more detailed with the --help parameter

Thx for that.

@glangford 
> To get help on all of the command line options, type whisper --help with a lowercase 'h' as @phineas-pta emphasized in his post. There is no more detailed documentation, other than the README, and the associated research paper. If you have a specific question about how to use whisper, I recommend also searching through the past discussions, there is a lot of useful information on different topics.

Ok, thx for that.

> No roadmap has been released for new developments in whisper. See these past discussion threads, there is an interest:

Does that mean the development of whisper is stopped?
Does this mean that the .pt model files are no longer updated?

Thanks for ur answers,
Regards

**Reply by glangford** [reply]
We don't know if development is stopped, or just paused temporarily, there has been no announcement. The last time the models were updated was in December (see discussion below).

* https://github.com/openai/whisper/discussions/661

**Reply by ryanheise** [reply]
Just a bit of speculation here: OpenAI has invested in Descript, a transcription-based video and audio editor, with an eventual goal of switching Descript's underlying transcription technology from Rev entirely to Whisper. Because Descript allows you to edit video and audio by editing the transcript, it actually requires very accurate transcript timestamps to work. For example, if you edit a word in the transcript, it will cut out that word from the audio, and use AI to generate the audio for the new word, then splice it right in seamlessly. You can also get it to automatically delete the Ums and Ahs. Whisper's current word timestamp accuracy has certainly improved, but I don't think it is quite at the level that would be needed for transcript-based audio editing, and so (again, speculation) I wouldn't be surprised if they were at least doing some fine tuning of the models to improve word timestamp accuracy in order to fulfill one of the goals of the Descript partnership.

**Reply by islogged** [reply]
> We don't know if development is stopped, or just paused temporarily, there has been no announcement. The last time the models were updated was in December (see discussion below).

Ok, Thank you for this informations...
Where are the announcements made ( when done ) ? so i could monitor them...

> Just a bit of speculation here: OpenAI has invested in Descript, a transcription-based video and audio editor, with an eventual goal of switching Descript's underlying transcription technology from Rev entirely to Whisper. Because Descript allows you to edit video and audio by editing the transcript, it actually requires very accurate transcript timestamps to work. For example, if you edit a word in the transcript, it will cut out that word from the audio, and use AI to generate the audio for the new word, then splice it right in seamlessly. You can also get it to automatically delete the Ums and Ahs. Whisper's current word timestamp accuracy has certainly improved, but I don't think it is quite at the level that would be needed for transcript-based audio editing, and so (again, speculation) I wouldn't be surprised if they were at least doing some fine tuning of the models to improve word timestamp accuracy in order to fulfill one of the goals of the Descript partnership.

Ok, anyway i think lot of people like me waiting for **an official option** to translate English to a non-English language, something like: --task translate targetCodeLanguage

Thx for All

**Comment by glangford** [comment]
> Is it planned to translate English to non-English soon? and is it requested by people as a new functionality ?

No roadmap has been released for new developments in whisper. See these past discussion threads, there is an interest:
* https://github.com/openai/whisper/discussions/649
* https://github.com/openai/whisper/discussions/566
* https://github.com/openai/whisper/discussions/1539
